src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -60,10 +64,28 @@ public JobService(IamService samService,
         this.appConfig = appConfig;
         this.stairwayJdbcConfiguration = stairwayJdbcConfiguration;
         this.migrateConfiguration = migrateConfiguration;
+        this.isShutdown = new AtomicBoolean(false);
 
         ExecutorService executorService = Executors.newFixedThreadPool(appConfig.getMaxStairwayThreads());
         ExceptionSerializer serializer = new StairwayExceptionSerializer(objectMapper);
-        stairway = new Stairway(executorService, applicationContext, serializer);
+        stairway = new Stairway(executorService, applicationContext, serializer, "stairwayname");
+    }
+
+    /**
+     * Stop accepting jobs and shutdown stairway
+     */
+    public void shutdown() throws InterruptedException {
+        isShutdown.set(true);
```

Comment:
Do you need a check of the atomic boolean at the beginning of this method to make sure that only one thread is calling shutdown() at a time?

Reply:
Good point. Yes, we should ignore duplicate shutdown requests.